x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FoldContext.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.core.expression;
+
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.xpack.esql.core.QlClientException;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+
+/**
+ * Context passed to {@link Expression#fold}.
+ */
+public class FoldContext {
+    /**
+     * {@link Expression#fold} using any amount of memory. Only safe for tests.
+     */
+    public static FoldContext unbounded() {
+        return new FoldContext(Long.MAX_VALUE);
```

Comment:
Since we are still using it in a few places, wouldn't it make sense to default it to the same 5% of the memory as the default in the query pragmas (or a slightly higher value), rather than making it really unbounded? It's a bit paranoid maybe, I guess the final goal here is safety

Reply:
I'm hopeful we can remove the usages in a follow-up. Let's get this in an have a conversation about whether or not we should replace `unbounded` with `small`.